saleor/graphql/attribute/types.py

```diff
@@ -656,3 +655,791 @@ class AttributeValueInput(BaseInputObjectType):
+        ),
+    )
+
+    class Meta:
+        interfaces = [AssignedAttribute]
+        description = "Represents multi page reference attribute." + ADDED_IN_322
+        doc_category = DOC_CATEGORY_ATTRIBUTES
+
+    @staticmethod
+    def resolve_value(
+        root: AssignedAttributeData,
+        info: ResolveInfo,
+        limit: int = DEFAULT_NESTED_LIST_LIMIT,
+    ) -> Promise[list[ChannelContext[page_models.Page]]]:
+        if not root.values:
+            return Promise.resolve([])
+
+        channel_slug = root.attribute.channel_slug
+        attr_values = [value.node for value in root.values]
+        attr_values = attr_values[:limit]
+
+        def _wrap_with_channel_context(
+            pages: list[page_models.Page],
+        ) -> list[ChannelContext[page_models.Page]]:
+            if not pages:
+                return []
+            return [
+                ChannelContext(node=page, channel_slug=channel_slug) for page in pages
+            ]
+
+        return (
+            PageByIdLoader(info.context)
+            .load_many([value.reference_page_id for value in attr_values])
+            .then(_wrap_with_channel_context)
+        )
+
+
+class AssignedMultiProductReferenceAttribute(BaseObjectType):
+    value = NonNullList(
+        "saleor.graphql.product.types.Product",
+        description="List of assigned product references.",
+        required=True,
+        limit=PositiveInt(
+            description=(
+                "Maximum number of referenced products to return. "
+                f"Default is {DEFAULT_NESTED_LIST_LIMIT}."
+            ),
+            default_value=DEFAULT_NESTED_LIST_LIMIT,
+        ),
+    )
+
+    class Meta:
+        interfaces = [AssignedAttribute]
+        description = "Represents multi product reference attribute." + ADDED_IN_322
+        doc_category = DOC_CATEGORY_ATTRIBUTES
+
+    @staticmethod
+    def resolve_value(
+        root: AssignedAttributeData,
+        info: ResolveInfo,
+        limit: int = DEFAULT_NESTED_LIST_LIMIT,
+    ) -> Promise[list[ChannelContext[product_models.Product]]]:
+        if not root.values:
+            return Promise.resolve([])
+
+        channel_slug = root.attribute.channel_slug
+        attr_values = [value.node for value in root.values]
+        attr_values = attr_values[:limit]
+
+        def _wrap_with_channel_context(
+            products: list[product_models.Product],
+        ) -> list[ChannelContext[product_models.Product]]:
+            if not products:
+                return []
+            return [
+                ChannelContext(node=product, channel_slug=channel_slug)
+                for product in products
+            ]
+
+        return (
+            ProductByIdLoader(info.context)
+            .load_many([value.reference_product_id for value in attr_values])
+            .then(_wrap_with_channel_context)
+        )
+
+
+class AssignedMultiProductVariantReferenceAttribute(BaseObjectType):
+    value = NonNullList(
```

Comment:
In this field, Saleor returns multiple values. 
```suggestion
    values = NonNullList(
```